mlir/lib/Dialect/SPIRV/IR/SPIRVTypes.cpp

```diff
@@ -175,7 +167,21 @@ void CompositeType::getCapabilities(
       .Case<VectorType>([&](VectorType type) {
         auto vecSize = getNumElements();
         if (vecSize == 8 || vecSize == 16) {
-          static const Capability caps[] = {Capability::Vector16};
+          static const Capability caps[] = {Capability::Vector16,
+                                            Capability::VectorAnyINTEL};
+          ArrayRef<Capability> ref(caps, std::size(caps));
+          capabilities.push_back(ref);
+        }
+        // If the vector size is between (2 - (2^63 - 1))
+        // and not of any size 2, 3, 4, 8, and 16
+        // VectorAnyIntel Capability must be present
+        // for the SPIR-V to be valid
+        llvm::SmallVector<uint32_t, 5> allowedVecRange = {2, 3, 4, 8, 16};
```

Comment:
This can be just `std::array`

Reply:
@Hardcode84 (Ivan),
Was trying to stick to llvm data structures as much as I could. 
Also  llvm::SmallVector doesn't do heap allocation until the size goes past the initial size, which is the case in here.

So shouldn't  llvm::SmallVector and std::array work the same in terms of performance?

Reply:
I don't think there will be any perf differences, but `std::array` (or just plain C array) better conveys the purpose as this is fixed size array.